crates/turbopack/src/condition.rs

```diff
@@ -27,12 +30,22 @@ impl ContextCondition {
         ContextCondition::Not(Box::new(condition))
     }
 
+    #[async_recursion]
     /// Returns true if the condition matches the context.
-    pub fn matches(&self, context: &FileSystemPath) -> bool {
+    pub async fn matches(&self, context: &FileSystemPath) -> bool {
```

Comment:
I think we tried to convert this to async before and saw a performance issue with that. Do you remember @alexkirsz ?

Reply:
If we change impl to `use something like InPath(FileSystemPathVc)`, how does match would work if we can't use async fn?

Reply:
Benchmark seem to be happy, so nevermind